src/audio/module_adapter/module_adapter.c

```diff
@@ -603,24 +615,36 @@ int module_adapter_copy(struct comp_dev *dev)
 	 * period and have only 1 source and 1 sink buffer
 	 */
 	if (mod->simple_copy) {
-		source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
 		sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-
-		source_c = buffer_acquire(source);
 		sink_c = buffer_acquire(sink);
-		comp_get_copy_limits_frame_aligned(source_c, sink_c, &c);
+		list_for_item(blist, &dev->bsource_list) {
+			source = container_of(blist, struct comp_buffer, sink_list);
+			source_c[i] = buffer_acquire(source);
 
-		buffer_stream_invalidate(source_c, c.frames * c.source_frame_bytes);
+			mod->input_buffers[i].size = 0;
 
-		/* note that the size is in number of frames not the number of bytes */
-		mod->input_buffers[0].size = c.frames;
-		mod->input_buffers[0].consumed = 0;
-		mod->output_buffers[0].size = 0;
+			/* check if the source dev is in the same state as the dev */
+			if (!source_c[i]->source && source_c[i]->source->state != dev->state)
+				continue;
```

Comment:
oh... that forces a NULL dereference. You meant `if (!source_c[i]->source || source_c[i]->source->state != dev->state)`